src/m365/spo/commands/file/file-sharinglink-add.ts

```diff
@@ -0,0 +1,188 @@
+      },
+      {
+        option: '-i, --fileId [fileId]'
+      },
+      {
+        option: '-f, --fileUrl [fileUrl]'
+      },
+      {
+        option: '--type <type>',
+        autocomplete: SpoFileSharingLinkAddCommand.type
+      },
+      {
+        option: '--expirationDateTime [expirationDateTime]'
+      },
+      {
+        option: '--scope [scope]',
+        autocomplete: SpoFileSharingLinkAddCommand.scope
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.fileId && !validation.isValidGuid(args.options.fileId)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        if (args.options.type &&
+          SpoFileSharingLinkAddCommand.type.indexOf(args.options.type) < 0) {
+          return `'${args.options.type}' is not a valid type. Allowed types are ${SpoFileSharingLinkAddCommand.type.join(', ')}`;
+        }
+
+        if (args.options.scope &&
+          SpoFileSharingLinkAddCommand.scope.indexOf(args.options.scope) < 0) {
+          return `'${args.options.scope}' is not a valid scope type. Allowed scope types are ${SpoFileSharingLinkAddCommand.scope.join(', ')}`;
+        }
+
+        const parsedDateTime = Date.parse(args.options.expirationDateTime as string);
+        if (args.options.expirationDateTime && !(!parsedDateTime) !== true) {
+          return `${args.options.expirationDateTime} is not a valid date format. Provide the date in one of the following formats:
+      ${chalk.grey('YYYY-MM-DD')}
+      ${chalk.grey('YYYY-MM-DDThh:mm')}
+      ${chalk.grey('YYYY-MM-DDThh:mmZ')}
+      ${chalk.grey('YYYY-MM-DDThh:mm±hh:mm')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['fileId', 'fileUrl'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Creates a sharing link for file ${args.options.fileId || args.options.fileUrl}...`);
+    }
+
+    try {
+      const fileDetails = await this.getFileDetails(args.options.webUrl, args.options.fileId, args.options.fileUrl);
+
+      const requestOptions: CliRequestOptions = {
+        url: `https://graph.microsoft.com/v1.0/sites/${fileDetails.SiteId}/drives/${fileDetails.VroomDriveID}/items/${fileDetails.VroomItemID}/createLink`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          type: args.options.type,
+          ...(args.options.expirationDateTime && { expirationDateTime: args.options.expirationDateTime }),
+          ...(args.options.scope && { scope: args.options.scope })
```

Comment:
Let's simplify these a bit

```suggestion
          expirationDateTime: args.options.expirationDateTime,
          scope: args.options.scope
```

If the value is undefined, it will be automatically excluded from the request.